doc/design/refactor/distributed_architecture.md

```diff
@@ -110,30 +112,88 @@ img, label = input[0], input[1]
 hidden = paddle.layer.fc(input=img, size=200, act=paddle.activation.Tanh())
 prediction = paddle.layer.fc(input=img, size=10, act=paddle.activation.Softmax())
 cost = paddle.layer.classification_cost(input=prediction, label=label)
-optimizer = paddle.optimizer.SGD(cost, learning_rate=0.01)
-session = paddle.session.NewRemote(num_trainer=3, num_ps=2, GPU_per_trainer=1)
+optimizer = paddle.optimizer.SGD(learning_rate=0.01)
+opts = optimizer.minimize(cost)
+exe = RemoteExecutor(num_trainer=3, num_ps=2, GPU_per_trainer=2, sync_batches=1)
```

Comment:
When I saw the name `RemoteExecutor`, I'd thought it might require a parameter of the remote IP...

I guess here you might want `DistributedExecutor`?

BTW, do we really need to expose the concept *executor* in our API to our users? I prefer not.

Reply:
Maybe `RemoteExecutor` should init with parameter servers' IP addresses, and `DistributedExecutor` use etcd to get server IPs.

> BTW, do we really need to expose the concept executor in our API to our users? I prefer not.

Thought `Executor` is the same concept of V2 API's `trainer` class, It's simple to wrap the name to let it compatible with v2 API.

Reply:
In my mind the `RemoteExecutor` and `DistributedExecutor` is a naming difference, we can use either one. We can put the IP (and authentication) as an argument, I initially thought the program will read the IP and authentication config from `~/.paddle/config`.

Reply:
I add some additional words.  We still hold the worldview that **Python code should be same when the user wants to switch between local and multi-nodes, multi-GPU**. 
The user should not aware the devices/nodes he may use, he only needs to care about the model part.  He only needs to modify the configuration file when switching between different environments. 

If we agree with that, so the `Remoteexecutor` should not be exposed, even the `executor` is also should be invisible.

Reply:
I have a concern about if we should add a device field in our framework.proto. For details, please refer to #6035. Sometimes, users do not want to aware the hardware environment. But sometimes, users do want to set device for specific operators. How can we handle this?

Reply:
@QiJune I agree, it's a very useful feature. Maybe we can work out a first version that the user can not do manual placement, but later add this feature?